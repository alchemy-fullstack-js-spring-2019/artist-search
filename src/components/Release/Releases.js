import React from 'react';
import PropTypes from 'prop-types';
import Release from './Release';

function Releases({ releaseArray, artist }) {
  const releaseList = releaseArray.map(release => {
    return (
      <li key={ release.id }>
        <Release
          id={ release.id }
          title={ release.title }
          date={ release.date }
          coverArt={ release.coverArt }
          artist={ artist }
        />
      </li>
    );
  });

  return (
    <ul>
      { releaseList }
    </ul>
  );
}

Releases.propTypes = {
  releaseArray: PropTypes.array.isRequired,
  artist: PropTypes.string.isRequired
};

export default Releases;
