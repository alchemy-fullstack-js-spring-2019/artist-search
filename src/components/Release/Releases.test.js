import React from 'react';
import { shallow } from 'enzyme';
import Releases from './Releases';

describe('Releases component', () => {
  it('renders the Releases', () => {
    const fakeData = [
      {
        id: '123',
        title: 'okay',
        date: '1942',
        coverArt: 'https://radiohala.com/wp-content/uploads/default-cover-art.png',
      },
      {
        id: '442',
        title: 'Nah',
        date: '1099',
        coverArt: 'https://radiohala.com/wp-content/uploads/default-cover-art.png1'
      }
    ];

    const wrapper = shallow(<Releases releaseArray={ fakeData } artist="Yes" />);
    expect(wrapper).toMatchSnapshot();
  });
});
