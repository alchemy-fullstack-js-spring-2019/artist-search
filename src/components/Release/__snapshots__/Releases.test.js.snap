// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Releases component renders the Releases 1`] = `
<ul>
  <li
    key="123"
  >
    <Release
      artist="Yes"
      coverArt="https://radiohala.com/wp-content/uploads/default-cover-art.png"
      date="1942"
      id="123"
      title="okay"
    />
  </li>
  <li
    key="442"
  >
    <Release
      artist="Yes"
      coverArt="https://radiohala.com/wp-content/uploads/default-cover-art.png1"
      date="1099"
      id="442"
      title="Nah"
    />
  </li>
</ul>
`;
